### PR TITLE
Fix #7636, add PEP 519 test coverage to remaining I/O functions

### LINKED ISSUE #7636: test_paths.TestPaths.test_whosmat failure on Python 3.6, 32-bit
```
======================================================================
FAIL: test_paths.TestPaths.test_whosmat
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/venv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/venv/lib/python3.6/site-packages/numpy/testing/decorators.py", line 147, in skipper_func
    return f(*args, **kwargs)
  File "/venv/lib/python3.6/site-packages/scipy/io/tests/test_paths.py", line 58, in test_whosmat
    assert_(contents[0] == ('data', (1, 5), 'int64'))
  File "/venv/lib/python3.6/site-packages/numpy/testing/utils.py", line 77, in assert_
    raise AssertionError(smsg)
AssertionError
```

Failing job: https://travis-ci.org/MacPython/scipy-wheels/jobs/255588602

Scipy wheel build matrix: https://travis-ci.org/MacPython/scipy-wheels/builds/255588592

### PR DESCRIPTION
**Pull request by mruffalo**
Apparently integers are stored as int32 on 32-bit Python, so just check the
name and shape of what's stored in the file.

### CODE REVIEW

File: scipy/io/tests/test_paths.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_whosmat(self):
             scipy.io.savemat(str(path), {'data': self.data})
 
             contents = scipy.io.whosmat(path)
-            assert_(contents[0] == ('data', (1, 5), 'int64'))
+            assert_(contents[0][0:2] == ('data', (1, 5)))
```
Comment (perimosocordiae) [inline scipy/io/tests/test_paths.py]: Add a comment here explaining why `contents[0][2]` isn't checked?

**Comment by pv** [comment]
Or "data = np.arange(5).astype(np.int64)" to avoid depending on the 
platform default integer width.

**Comment by mruffalo** [comment]
Both sound reasonable to me. I had already started to add a comment to that assertion explaining why the data type is omitted, and I was about to push an amended commit.

In principle we don't even need to check the result of `whosmat` here; I just added that because it didn't seem to hurt (which of course backfired with different integer widths on different platforms). What really matters here is that `whosmat` doesn't throw an exception when given a `Path` object, and checking that it can read something written by `savemat` isn't necessarily within the scope of this test.

**Comment by perimosocordiae** [comment]
I like @pv's suggestion to fix the dtype of `data`. It's not the purpose of this test, but it makes things clearer.

The TravisCI failures are unrelated.

**Comment by mruffalo** [comment]
The type of data is now specified explicitly in `test_paths.py` as of c2fc0b2.

I added two follow-up commits that test the rest of the I/O functions I mentioned in my final comment in #6855. None of these needed any code changes -- just adding test coverage to ensure that `Path` objects continue to work in the future.

**Pull request merged**